### ISSUE TITLE
surprising results from `revolve()`

### ISSUE THREAD

**Issue by wrongbad** [elalish/manifold#672]
In the first example, I would expect a normal sphere, but it seems to apply the "only positive x plane" constraint by dropping vertices rather than clipping the polygon, and then vertices where x==0 are inconsistent.

The second example (revolving only 180 degrees) is weirder, because the behavior above still occurs for the "outer" surface, but not for the "inner" surface.

```py
from manifold3d import *
set_circular_segments(16)
m = CrossSection.circle(radius=10).revolve()
```

<img width="413" alt="Screen Shot 2023-12-26 at 4 25 33 PM" src="https://github.com/elalish/manifold/assets/10201449/55bebb0a-8ca7-403b-9030-bd1254a48b8a">

```py
from manifold3d import *
set_circular_segments(16)
m = CrossSection.circle(radius=10).revolve(revolve_degrees=180)
```

<img width="501" alt="Screen Shot 2023-12-26 at 4 26 49 PM" src="https://github.com/elalish/manifold/assets/10201449/f7c67ee8-b3c8-4701-b46a-7c6f85b6bd35">

**Comment by elalish** [comment]
Weird, I could swear that used to work. Apparently we need more tests. Thanks!

**Comment by elalish** [comment]
I have more things in the works, but I'm making slow progress, as I'm still on vacation. I'm thinking I should just get a fix in for this bug instead and put out a v2.3 release. What do you think @pca006132?

**Comment by pca006132** [comment]
I think it is fine, we have enough stuff for a release.
